packages/session-replay-browser/src/session-replay.ts

```diff
@@ -318,20 +334,24 @@ export class SessionReplay implements AmplitudeSessionReplay {
     return maskSelector as unknown as string;
   }
 
-  recordEvents() {
-    const shouldRecord = this.getShouldRecord();
+  captureEventsIfShould() {
+    const shouldRecord = this.getShouldCapture();
     const sessionId = this.identifiers?.sessionId;
     if (!shouldRecord || !sessionId || !this.config) {
       return;
     }
-    this.stopRecordingEvents();
+    if (this.recordCancelCallback) {
+      this.loggerProvider.debug('captureEvents method fired - Session Replay capture already in progress.');
+      return;
+    }
```

Comment:
Maybe this is worth putting at the top of the method? It seems like we'll always want to exit early if we're already capturing?